base_layer/common_types/src/emoji.rs

```diff
@@ -97,7 +98,7 @@ lazy_static! {
     pub static ref REVERSE_EMOJI: HashMap<char, u8> = {
         let mut m = HashMap::with_capacity(DICT_SIZE);
         EMOJI.iter().enumerate().for_each(|(i, c)| {
-            m.insert(*c, i as u8);
+            m.insert(*c, u8::try_from(i).expect("Invalid emoji"));
```

Comment:
Since the current table size ensures this case is safe, it seems like you could remove the panic and instead use a `debug_assert` where the table constants are defined.